### PR TITLE
Added DynamicIntArray.h/cpp and HW19 files

### PR DESCRIPTION
**Pull request by shintanshin**
Please review my 19th homework

### CODE REVIEW

**Review by Nothing-avil — APPROVED** [review]

File: HW19/DynamicIntArray.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    // Allocate new memory for new element
+    // Copy old content to new inner array
+    // insert element at the last index
+    // update size
+    void push_back(int element);
+
+private:
+    int* m_data;
+    std::size_t m_size;
```
Comment (bohdanpc) [inline HW19/DynamicIntArray.h:39]: glad to see you didn't get code from the same source as bunch of other students did)

File: HW19/DynamicIntArray.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		for (std::size_t i = 0; i < m_size; ++i) {
+			m_data[i] = other.m_data[i];
+		}
+	}
+	return *this;
+}
+
+int& DynamicIntArray::operator[](std::size_t index)
+{
+	return m_data[index];
```
Comment (bohdanpc) [inline HW19/DynamicIntArray.cpp:44]: it's best to add some validation for the _DEBUG version

Comment (bohdanpc) [reply HW19/DynamicIntArray.cpp:44]: or warning

File: HW19/DynamicIntArray.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+void DynamicIntArray::setSize(std::size_t newSize)
+{
+	int* newData = new int[newSize];
+	std::size_t minSize = (newSize < m_size) ? newSize : m_size;
```
Comment (bohdanpc) [inline HW19/DynamicIntArray.cpp:50]: const

File: HW19/DynamicIntArray.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	delete[] m_data;
+	m_data = newData;
+	m_size = newSize;
+}
+
+std::size_t DynamicIntArray::getSize() const
+{
+	return m_size;
+}
```
Comment (bohdanpc) [inline HW19/DynamicIntArray.cpp:64]: it's ok to put in header

**Pull request merged**